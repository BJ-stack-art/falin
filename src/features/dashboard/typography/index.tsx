import {
  Box,
  Card,
  Divider,
  Grid,
  Stack,
  SxProps,
  Typography,
  TypographyVariant,
} from "@mui/material";
import React from "react";

const listTypograpy: TypographyVariant[] = [
  "h1",
  "h2",
  "h3",
  "h4",
  "h5",
  "h6",
  "subtitle1",
  "subtitle2",
  "body1",
  "body2",
  "caption",
  "overline",
];

const styles: { card: SxProps } = {
  card: {
    padding: "20px",
    boxShadow: 3,
  },
};
const TypographyList: React.FC = () => {
  return (
    <Box>
      <Typography variant="h6">Typography</Typography>
      <Grid container mt={4} spacing={1}>
        <Grid item md={3}>
          <Card sx={styles.card}>
            <Typography variant="h6">Regular</Typography>
            <Divider sx={{ mt: 1, mb: 2 }} />
            <Stack direction={"column"}>
              {listTypograpy.map((variant, i) => (
                <Typography variant={variant} key={i} fontWeight="regular">
                  Text {variant}
                </Typography>
              ))}
            </Stack>
          </Card>
        </Grid>
        <Grid item md={3}>
          <Card sx={styles.card}>
            <Typography variant="h6">Medium</Typography>
            <Divider sx={{ mt: 1, mb: 2 }} />
            <Stack direction={"column"}>
              {listTypograpy.map((variant, i) => (
                <Typography variant={variant} key={i} fontWeight="medium">
                  Text {variant}
                </Typography>
              ))}
            </Stack>
          </Card>
        </Grid>
        <Grid item md={3}>
          <Card sx={styles.card}>
            <Typography variant="h6">Semi Bold</Typography>
            <Divider sx={{ mt: 1, mb: 2 }} />
            <Stack direction={"column"}>
              {listTypograpy.map((variant, i) => (
                <Typography variant={variant} key={i} fontWeight="semiBold">
                  Text {variant}
                </Typography>
              ))}
            </Stack>
          </Card>
        </Grid>
        <Grid item md={3}>
          <Card sx={styles.card}>
            <Typography variant="h6">Bold</Typography>
            <Divider sx={{ mt: 1, mb: 2 }} />
            <Stack direction={"column"}>
              {listTypograpy.map((variant, i) => (
                <Typography variant={variant} key={i} fontWeight="bold">
                  Text {variant}
                </Typography>
              ))}
            </Stack>
          </Card>
        </Grid>
      </Grid>
    </Box>
  );
};

export default TypographyList;
