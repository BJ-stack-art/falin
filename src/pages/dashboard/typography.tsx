import { CircularProgress } from "@mui/material"
import React, { lazy, Suspense } from "react"
const TypographyList = lazy(() => import("@/features/dashboard/typography"))

const TypographyPage: React.FC = () => {
  return (
    <Suspense fallback={<CircularProgress />}>
      <TypographyList />
    </Suspense>
  )
}

export default TypographyPage
