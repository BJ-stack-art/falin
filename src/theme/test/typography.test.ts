import { describe, expect, test } from "vitest"
import { typography } from "../typography"

describe("Theme Typography", () => {
  test("should match snapshot", () => {
    expect(typography).toMatchSnapshot()
  })
})
