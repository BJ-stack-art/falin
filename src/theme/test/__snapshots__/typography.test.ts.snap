// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Theme Typography > should match snapshot 1`] = `
{
  "body1": {
    "fontSize": "1rem",
    "lineHeight": "1.5rem",
  },
  "body2": {
    "fontSize": "0.875rem",
    "lineHeight": "1.25rem",
  },
  "caption": {
    "fontSize": "0.75rem",
    "lineHeight": "1.125rem",
  },
  "fontFamily": "Inter",
  "fontSize": 16,
  "fontWeightBold": 700,
  "fontWeightMedium": 500,
  "fontWeightRegular": 400,
  "fontWeightSemiBold": 600,
  "h1": {
    "fontSize": "4.5rem",
    "letterSpacing": "-2%",
    "lineHeight": "5.625rem",
  },
  "h2": {
    "fontSize": "3.75rem",
    "letterSpacing": "-2%",
    "lineHeight": "4.5rem",
  },
  "h3": {
    "fontSize": "3rem",
    "letterSpacing": "-2%",
    "lineHeight": "3.75rem",
  },
  "h4": {
    "fontSize": "2.25rem",
    "letterSpacing": "-2%",
    "lineHeight": "2.75rem",
  },
  "h5": {
    "fontSize": "1.875rem",
    "lineHeight": "2.375rem",
  },
  "h6": {
    "fontSize": "1.5rem",
    "lineHeight": "2rem",
  },
  "overline": {
    "fontSize": "0.625rem",
    "lineHeight": "1rem",
  },
  "subtitle1": {
    "fontSize": "1.25rem",
    "lineHeight": "1.875rem",
  },
  "subtitle2": {
    "fontSize": "1.125rem",
    "lineHeight": "1.75rem",
  },
}
`;
